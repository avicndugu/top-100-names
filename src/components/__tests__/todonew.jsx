import { render, screen, cleanup } from '@testing-library/react';
import TodoNew from '../todonew';

afterEach(() => {
  cleanup();
});

test('should render non-completed Todonew', () => {
  const todo = { id: 1, title: 'wash dishes', completed: false, };
  render(<TodoNew todo={todo} />)
  const todoElement = screen.getByTestId('todo-1');
  expect(todoElement).toBeInTheDocument();
  expect(todoElement).toHaveTextContent('wash dishes');
  expect(todoElement).not.toContainHTML('del');
})

test('should render completed Todonew', () => {
  const todo = { id: 2, title: 'wash pillow', completed: true, };
  render(<TodoNew todo={todo} />)
  const todoElement = screen.getByTestId('todo-2');
  expect(todoElement).toBeInTheDocument();
  expect(todoElement).toHaveTextContent('wash pillow');
  expect(todoElement).toContainHTML('del');
  // or
  expect(todoElement).toMatchSnapshot('<del>');

})