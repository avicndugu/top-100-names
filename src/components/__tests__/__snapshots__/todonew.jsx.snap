// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render completed Todonew: <del> 1`] = `
<div
  data-testid="todo-2"
>
  <del>
    <h6>
      wash pillow
    </h6>
  </del>
</div>
`;
